crates/containerd-shim-wasm/src/libcontainer_instance/container_executor.rs

```diff
@@ -32,7 +32,19 @@ impl Executor for LinuxContainerExecutor {
         self.default_executor.exec(spec)
     }
 
+    fn validate(&self, spec: &Spec) -> Result<(), ExecutorValidationError> {
+        self.default_executor.validate(spec)?;
+
+        self.can_handle(spec)
+            .then_some(())
+            .ok_or(ExecutorValidationError::InvalidArg)
+    }
+}
+
+impl LinuxContainerExecutor {
     fn can_handle(&self, spec: &Spec) -> bool {
+        // TODO: some of these logic is now implemented inside the default
```

Comment:
We should address this TODO before merging.

Reply:
I think it's ok to completely remove this function now.